src/osd/scrubber/scrub_backend.cc

```diff
@@ -1817,21 +1827,78 @@ std::optional<snap_mapper_fix_t> ScrubBackend::scan_object_snaps(
   }
   set<snapid_t> obj_snaps{p->second.begin(), p->second.end()};
 
-  set<snapid_t> cur_snaps;
-  int r = snaps_getter.get_snaps(hoid, &cur_snaps);
-  if (r != 0 && r != -ENOENT) {
-    derr << __func__ << ": get_snaps returned " << cpp_strerror(r) << dendl;
-    ceph_abort();
+  // check/fix snapset. Should match what we have in the object.
+  auto cur_snaps = m_pool.info.is_tier()
+		     ? snaps_getter.get_snaps(hoid)
+		     : snaps_getter.get_verified_snaps(hoid);
```

Comment:
Why can't we validate the snapid->hobject mapping for a tier?

Reply:
Discussed in some meetings I had the last couple of days. Let's go over that in the Q&A meeting today.

Reply:
For posterity: following a discussion with Sam:

- tiering is pool-based - the one "logical pool" is serviced by the base pool (the full one), and a cache pool;
- the SnapMapper DBs are fully "internally consistent" in both pools;
- the Scrubber should (and does) know, when handling the cache pool, to expect that specific clones would not exist;
- there will be a clear cutoff - if a clone exist, so would the younger ones.

(any errors in the phrasing above are mine...)